### PR TITLE
[Refactor] refactor the issue filing button component to take optional renderer

### PR DESCRIPTION
**Pull request by haonliu**
#### Description of changes

refactor the issue filing button component to have the option to show help text

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Added relevant unit test for your changes. (`npm run test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`npm run precheckin`)
- [ ] Added screenshots/GIFs for UI changes.

### CODE REVIEW

File: src/common/components/issue-filing-button.tsx [hunk 1]
```diff
@@ -5,12 +5,16 @@ import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
...
-import { IssueFilingDialog, IssueFilingDialogDeps } from '../../DetailsView/components/issue-filing-dialog';
+import { IssueFilingDialogDeps } from '../../DetailsView/components/issue-filing-dialog';
 import { EnvironmentInfo } from '../environment-info-provider';
 import { LadyBugSolidIcon } from '../icons/lady-bug-solid-icon';
 import { BugActionMessageCreator } from '../message-creators/bug-action-message-creator';
 import { FileIssueClickService } from '../telemetry-events';
 import { CreateIssueDetailsTextData } from '../types/create-issue-details-text-data';
+import {
+    IssueFilingNeedsSettingsContentRenderer,
+    IssueFilinigNeedsSettingsContentProps,
```
Comment (peterdur) [inline src/common/components/issue-filing-button.tsx]: Typo
```suggestion
    IssueFilingNeedsSettingsContentProps,
```

File: src/common/components/issue-filing-button.tsx [hunk 2]
```diff
@@ -49,6 +54,17 @@ export class IssueFilingButton extends React.Component<IssueFilingButtonProps, I
             : null;
         const target: string = isSettingValid ? '_blank' : '_self';
 
+        const needsSettingsContentProps: IssueFilinigNeedsSettingsContentProps = {
```
Comment (peterdur) [inline src/common/components/issue-filing-button.tsx]: `IssueFilinigNeedsSettingsContentProps` only has property `isOpen` but there are lots of other properties below; this seems strange.

File: src/DetailsView/components/issue-filing-dialog.tsx [hunk 3]
```diff
@@ -8,10 +8,11 @@ import { BugFilingService } from '../../bug-filing/types/bug-filing-service';
 import { EnvironmentInfoProvider } from '../../common/environment-info-provider';
 import { NamedSFC } from '../../common/react/named-sfc';
 import { CreateIssueDetailsTextData } from '../../common/types/create-issue-details-text-data';
+import { IssueFilinigNeedsSettingsContentProps } from '../../common/types/issue-filing-needs-setting-content';
 import { BugServiceProperties } from '../../common/types/store-data/user-configuration-store';
 import { ActionAndCancelButtonsComponent } from './action-and-cancel-buttons-component';
 
-export interface IssueFilingDialogProps {
+export interface IssueFilingDialogProps extends IssueFilinigNeedsSettingsContentProps {
```
Comment (peterdur) [inline src/DetailsView/components/issue-filing-dialog.tsx]: I wonder if this is backwards - should it be that `IssueFilinigNeedsSettingsContentProps` extends `IssueFilingDialogProps`? (see comment at line 57 below)

Comment (haonliu) [reply src/DetailsView/components/issue-filing-dialog.tsx]: right, Make sense!

File: src/common/components/needs-settings-help-text.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import * as React from 'react';
+
+import { NamedSFC } from '../react/named-sfc';
+
+export type IssueFilinigNeedsSettingsHelpTextProps = {
```
Comment (peterdur) [inline src/common/components/needs-settings-help-text.tsx]: Typo (Filinig should be Filing)
```suggestion
export type IssueFilingNeedsSettingsHelpTextProps = {
```

File: src/common/components/needs-settings-help-text.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    isOpen: boolean;
+};
+
+export const IssueFilinigNeedsSettingsHelpText = NamedSFC<IssueFilinigNeedsSettingsHelpTextProps>('IssueFilingDialog', props => {
```
Comment (peterdur) [inline src/common/components/needs-settings-help-text.tsx]: Typo x 2:
```suggestion
export const IssueFilingNeedsSettingsHelpText = NamedSFC<IssueFilingNeedsSettingsHelpTextProps>('IssueFilingDialog', props => {
```

File: src/tests/unit/tests/common/components/issue-filing-needs-settings-help-text.test.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+// Licensed under the MIT License.
+import { shallow } from 'enzyme';
+import * as React from 'react';
+
+import {
+    IssueFilinigNeedsSettingsHelpText,
+    IssueFilinigNeedsSettingsHelpTextProps,
+} from '../../../../../common/components/needs-settings-help-text';
+
+describe('IssueFilinigNeedsSettingsHelpTextTest', () => {
```
Comment (peterdur) [inline src/tests/unit/tests/common/components/issue-filing-needs-settings-help-text.test.tsx]: Typo:
```suggestion
describe('IssueFilingNeedsSettingsHelpTextTest', () => {
```

File: src/tests/unit/tests/common/components/__snapshots__/issue-filing-needs-settings-help-text.test.tsx.snap [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+exports[`IssueFilingNeedsSettingsHelpTextTest render with isOpen = true 1`] = `
+<div
+  aria-live="polite"
+  className="create-bug-button-help"
+  role="alert"
+>
+  Go to Settings to configure issue filing.
+</div>
+`;
+
```
Comment (peterdur) [inline src/tests/unit/tests/common/components/__snapshots__/issue-filing-needs-settings-help-text.test.tsx.snap]: Are the snapshots on lines 15-25 obsolete versions from before the Filinig typo was fixed?

Comment (haonliu) [reply src/tests/unit/tests/common/components/__snapshots__/issue-filing-needs-settings-help-text.test.tsx.snap]: Right, I just updated.

**Pull request merged**